### PR TITLE
Generate appropriate LLVM IR attributes for custom optimization modes set on a SILFunction

### PR DESCRIPTION
**Pull request by meg-gupta**
Fixes rdar://113584125

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Review by aschwaighofer — APPROVED** [review]

File: test/IRGen/optmode.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    return n * n
+}
+
+// CHECK:@"$s7optmode7square21nS2i_tF"{{.*}}[[ATTR2:#[0-9]+]]
+@_optimize(size)
+func square2(n: Int) -> Int {
+    return n * n
+}
+
+// attributes [[ATTR1]] = { {{.*}}noinline optnone{{.*}} }
```
Comment (tshortli) [inline test/IRGen/optmode.swift:15]: Should these be `CHECK:` lines?

Comment (meg-gupta) [reply test/IRGen/optmode.swift]: Yess! Thank you

**Comment by meg-gupta** [comment]
@swift-ci smoke test and merge

**Pull request merged**